amundsen_application/static/js/ducks/search/types.ts

```diff
@@ -1,74 +1,64 @@
 import {
+  DashboardResource,
   Resource,
   ResourceType,
-  DashboardResource,
+  SearchAllOptions,
   TableResource,
   UserResource,
 } from 'interfaces';
-import { SearchReducerState } from './reducer';
 
 interface SearchResults<T extends Resource> {
   page_index: number;
   total_results: number;
   results: T[];
-}
-
+};
 export type DashboardSearchResults = SearchResults<DashboardResource>;
 export type TableSearchResults = SearchResults<TableResource>;
 export type UserSearchResults = SearchResults<UserResource>;
-export type SearchResponse = {
-  msg: string;
-  status_code: number;
-  search_term: string;
-  dashboards?: DashboardSearchResults;
-  tables?: TableSearchResults;
-  users?: UserSearchResults;
-}
 
-/* searchAll - Search all resource types */
 export enum SearchAll {
-  ACTION = 'amundsen/search/SEARCH_ALL',
+  REQUEST = 'amundsen/search/SEARCH_ALL_REQUEST',
   SUCCESS = 'amundsen/search/SEARCH_ALL_SUCCESS',
   FAILURE = 'amundsen/search/SEARCH_ALL_FAILURE',
   RESET = 'amundsen/search/SEARCH_ALL_RESET',
-}
-
-export interface SearchAllOptions {
-  dashboardIndex?: number;
-  tableIndex?: number;
-  userIndex?: number;
-}
-
+};
 export interface SearchAllRequest {
   options: SearchAllOptions;
   term: string;
-  type: SearchAll.ACTION;
-}
-
+  type: SearchAll.REQUEST;
```

Comment:
Any thoughts on adding `payload` to requests?

Reply:
Yeah called that out in https://github.com/lyft/amundsenfrontendlibrary/pull/196. Will move forward with that. (still in a separate PR though)

Reply:
Addressed in #197.